ocaml-lsp-server/src/client.ml

```diff
@@ -0,0 +1,54 @@
+open Import
+
+module Vscode = struct
+  (** A collection of VS Code editor commands.
+
+      Reference for VS Code built-in commands:
+      https://code.visualstudio.com/api/references/commands *)
+  module Commands = struct
+    (** [editor.action.triggerSuggest] is a vscode-specific command, which
+        triggers the completion request on all completion providers *)
+    let triggerSuggest =
+      Command.create ~title:"Trigger Suggest"
+        ~command:"editor.action.triggerSuggest" ()
+
+    (** Represents custom commands, i.e., commands added by an extension.
+
+        Currently, the module includes custom commands introduced by "Ocaml
+        Platform" extension *)
+    module Custom : sig
+      (** Request client cursor to jump to the next hole. Looks for a hole
+          starting at position [start_position], if provided; otherwise, uses
+          the cursor position. Will not show a pop-up notification if
+          [notify-if-no-hole] is set to [false] (the default value is [true]) *)
+      val next_hole :
+           ?start_position:Position.t
+        -> ?notify_if_no_hole:bool
```

Comment:
As far as I can tell, this argument is only passed with `false`.
I didn't look closely at `start_position` but its "optionality" is also suspect. Please make it required unless there's strong justification for it.

Reply:
This function aims to replicate 1-1 the interface provided by that command, so those configs are optional as in the command's interface. It is true that we now usually use both of these optional commands, but that may not be true in the future, and for example, a new comer will find this function's new interface puzzling or limiting. So my goal here was to keep 1-1 interface with the command. Does this make a strong justification? 

---

A note about the command: `ocaml.next-hole` is a command introduced by vscode-ocaml-platform, that has optional (both) arguments "position" and "notify-if-no-hole". They both have default values, where "position" defaults to the current position of the cursor, and "notify-if-no-hole" defaults to `true`.

Reply:
I see two different concerns here:

* Future proofing the need to use the full interface vscode provides us. I'm not a fan, but I can let it be if there's some vague plans to actually use this stuff in the future.

* Using optional arguments in the API. Let's consider the newcomer you mentioned, should we make it easy for him to avoid thinking about which arguments to provide when using functionality? It seems like an easy way to introduce bugs.

> They both have default values, where "position" defaults to the current position of the cursor,

That is a concerning API then. I would much prefer if the current position was an explicit value.

> and "notify-if-no-hole" defaults to true.

I do not see what is gained by not providing the intended value every time.